dipy/direction/ptt_direction_getter.pyx

```diff
@@ -0,0 +1,472 @@
+                                               OUTSIDEIMAGE,
+                                               INVALIDPOINT)
+from dipy.tracking.utils import min_radius_curvature_from_angle
+
+
+cdef class PTTDirectionGetter(ProbabilisticDirectionGetter):
+    """Parallel Transport Tractography (PTT) direction getter.
+    """
+    cdef double       angular_separation
+    cdef double       data_support_exponent
+    cdef double[3][3] frame
+    cdef double       k1
+    cdef double       k2
+    cdef double       k_small
+    cdef double       last_val
+    cdef double       last_val_cand
+    cdef double       max_angle
+    cdef double       max_curvature
+    cdef double[3]    position
+    cdef int          probe_count
+    cdef double       probe_length
+    cdef double       probe_normalizer
+    cdef int          probe_quality
+    cdef double       probe_radius
+    cdef double       probe_step_size
+    cdef double[9]    propagator
+    cdef double       step_size
+
+
+    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=None,
+                 double probe_length=0.5, double probe_radius=0,
+                 int probe_quality=3, int probe_count=1,
+                 double data_support_exponent=1, **kwargs):
+        """PTT used probe for estimating future propagation steps. A probe is a
+        short, cylinderical model of the connecting segment.
+
+        Parameters
+        ----------
+        pmf_gen : PmfGen
+            Used to get probability mass function for selecting tracking
+            directions.
+        max_angle : float, [0, 90]
+            Is used to set the upper limits for the k1 and k2 parameters
+            of parallel transport frame (max_curvature).
+        sphere : Sphere
+            The set of directions to be used for tracking.
+        pmf_threshold : float, [0., 1.]
+            Used to remove direction from the probability mass function for
+            selecting the tracking direction.
+        probe_length : double
+            The length of the probes. Shorter probe_length yields more
+            dispersed fibers.
+        probe_radius : double
+            The radius of the probe. A large probe_radius helps mitigate noise
+            in the pmf but it might make it harder to sample thin and intricate
+            connections, also the boundary of fiber bundles might be eroded.
+        probe_quality : integer,
+            The quality of the probe. This parameter sets the number of
+            segments to split the cylinder along the length of the
+            probe (minimum=2).
+        probe_count : integer
+            The number of probes. This parameter sets the number of parallel
+            lines used to model the cylinder (minimum=1).
+        data_support_exponent : double
+            Data support to the power dataSupportExponent is used for
+            rejection sampling.
+
+         """
+        if probe_length <= 0:
```

Comment:
can it be 0 really ?

Reply:
It cannot be, and will trigger the `ValueError`. Similarly, maybe it would be clearer to have `if not probe_lenght > 0`. What do you think?

Reply:
yes, it would be clearer